Doc/library/os.rst

```diff
@@ -798,18 +798,31 @@ as internal buffering of data.
    Copy *count* bytes from file descriptor *src*, starting from offset
    *offset_src*, to file descriptor *dst*, starting from offset *offset_dst*.
    If *offset_src* is None, then *src* is read from the current position;
-   respectively for *offset_dst*. The files pointed by *src* and *dst*
+   respectively for *offset_dst*.
+
+   In Linux kernel older than 5.3, the files pointed by *src* and *dst*
    must reside in the same filesystem, otherwise an :exc:`OSError` is
    raised with :attr:`~OSError.errno` set to :data:`errno.EXDEV`.
 
    This copy is done without the additional cost of transferring data
    from the kernel to user space and then back into the kernel. Additionally,
-   some filesystems could implement extra optimizations. The copy is done as if
+   some filesystems could implement extra optimizations, such as the use of
+   reflinks (i.e., two or more inodes that share pointers to the same
+   copy-on-write disk blocks; supported file systems include btrfs and XFS)
+   and server-side copy (in the case of NFS). The copy is done as if
    both files are opened as binary.
 
    The return value is the amount of bytes copied. This could be less than the
    amount requested.
 
+   .. note::
+
+      This function should not be used for copying a file from special
+      filesystems like procfs and sysfs because of
+      `a known issue <https://lore.kernel.org/linux-fsdevel/20210126135012.1.If45b7cdc3ff707bc1efa17f5366057d60603c45f@changeid/>`_.
```

Comment:
The link contains a fix. Was the fix merged into Linux?

Reply:
No, it wasn't. I tested `os.copy_file_range` for copying `/proc/cpuinfo` on Linux 5.17, it returned 0 and the destination file was empty.

Reply:
BTW, some Python functions that operate files are prone to similar issues too :(. I'll create a few bug reports.

Reply:
> BTW, some Python functions that operate files are prone to similar issues too :(. I'll create a few bug reports.

#93300 and #93296 are two examples.

Reply:
I would prefer to not add a link to an email.

Reply:
What do you think about such a note?

> :func:`os.copy_file_range` should not be used for copying a range of a pseudo file from a special filesystem like procfs and sysfs. It will always copy no bytes and return 0 as if the file was empty because of a known Linux kernel issue.